src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCacheItem.cs

```diff
@@ -38,7 +39,7 @@ public sealed class TokenCacheItem
         /// <summary>
         /// Default constructor.
         /// </summary>
-        internal TokenCacheItem(TokenCacheKey key, AuthenticationResult result)
+        internal TokenCacheItem(TokenCacheKey key, AdalResult result)
```

Comment:
Since you're passing in AdalResult, you can do the same thing you did with AuthenticationResult: ```_adalResult = adalResult ?? throw new ArgumentNullException();
            UserInfo = new UserInfo(_adalResult.UserInfo);```
And then set the properties like you did in AuthenticationResult ```public string AccessTokenType => _adalResult.AccessTokenType;```